calico-vpp-agent/routing/connectivity/ipip.go

```diff
@@ -57,6 +59,26 @@ func (p *IpipProvider) RescanState() {
 			p.ipipIfs[tunnel.Dst.String()] = tunnel
 		}
 	}
+
+	p.log.Infof("Rescanning existing routes")
+	p.ipipRoutes = make(map[uint32]map[string]bool)
+	routes, err := p.vpp.GetRoutes(0, false)
+	if err != nil {
+		p.log.Errorf("Error listing routes: %v", err)
+	}
+	for _, route := range routes {
+		for _, routePath := range route.Paths{
+			for _, tunnel := range p.ipipIfs{
```

Comment:
Might be worth adding a temporary `map[uint32]Tunnel` that we would fill in the loop above 
to avoid the nested loop.